src/Plugin/Field/FieldFormatter/AdvancedEntityReferenceFormatter.php

```diff
@@ -38,48 +40,95 @@ public static function defaultSettings() {
   public function settingsForm(array $form, FormStateInterface $form_state) {
     $elements = parent::settingsForm($form, $form_state);
     $elements['view_mode']['#title'] = $this->t('Default view mode');
+    $target_type = $this->getFieldSetting('target_type');
+    $handler_settings = $this->getFieldSetting('handler_settings');
+    $target_bundles = $handler_settings['target_bundles'];
+    $all_view_modes = $this->entityDisplayRepository->getViewModeOptions($target_type);
+    $field_name = $this->fieldDefinition->getName();
+    $settings = $this->getSettings();
+    $elements['override'] = [
+      '#type' => 'radios',
+      '#options' => [
+        'none' => $this->t('None'),
+        'bundle' => $this->t('Select view modes based on target bundle'),
+        'delta' => $this->t('Select view modes based on item delta'),
+      ],
+      '#title' => $this->t('Override options'),
+      '#default_value' => $settings['override'],
+    ];
+    $elements['bundle_view_mode'] = [
+      '#type' => 'container',
+      '#attributes' => ['id' => 'bundle-view-mode-wrapper'],
+      '#states' => [
+        'visible' => [
+          ':input[name="fields[' . $field_name . '][settings_edit_form][settings][override]"]' => ['value' => 'bundle'],
+        ],
+      ],
+    ];
+    foreach ($target_bundles as $bundle) {
+      // Get view modes for current bundle only.
+      $bundle_view_modes = $this->entityDisplayRepository->getViewModeOptionsByBundle($target_type, $bundle);
+      $elements['bundle_view_mode'][$bundle] = [
+        '#type' => 'select',
+        '#options' => $bundle_view_modes,
+        '#title' => $this->t('View mode for bundle %bundle', ['%bundle' => $bundle]),
+        '#default_value' => $settings['bundle_view_mode'][$bundle],
```

Comment:
`$settings['bundle_view_mode]` is an empty array by default so this will throw warnings. Need to validate that `$bundle` key exists.

Reply:
fixed